# Titanic competition with TensorFlow Decision Forests

This notebook will take you through the steps needed to train a baseline Gradient Boosted Trees Model using TensorFlow Decision Forests and creating a submission on the Titanic competition. 

This notebook shows:

1. How to do some basic pre-processing. For example, the passenger names will be tokenized, and ticket names will be splitted in parts.
1. How to train a Gradient Boosted Trees (GBT) with default parameters
1. How to train a GBT with improved default parameters
1. How to tune the parameters of a GBTs
1. How to train and ensemble many GBTs

# Imports dependencies

In [1]:
import numpy as np
import pandas as pd
import os

import tensorflow as tf
import tensorflow_decision_forests as tfdf
#Disable warning
import warnings
warnings.filterwarnings("ignore")

print(f"Found TF-DF {tfdf.__version__}")

2023-11-14 11:28:57.071976: I tensorflow/tsl/cuda/cudart_stub.cc:28] Could not find cuda drivers on your machine, GPU will not be used.
2023-11-14 11:28:57.353296: I tensorflow/tsl/cuda/cudart_stub.cc:28] Could not find cuda drivers on your machine, GPU will not be used.
2023-11-14 11:28:57.359947: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-11-14 11:29:00.990063: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT


Found TF-DF 1.5.0


# Load dataset

In [2]:
try:
    train_df = pd.read_csv("Dataset/train_df_age.csv")
    serving_df = pd.read_csv("Dataset/test.csv")
except:
    train_df = pd.read_csv("/kaggle/input/titanic/train_df_age.csv")
    serving_df = pd.read_csv("/kaggle/input/titanic/test.csv")

train_df.head(10)


Unnamed: 0  PassengerId  Survived  Pclass  \
0           0            1         0       3   
1           1            2         1       1   
2           2            3         1       3   
3           3            4         1       1   
4           4            5         0       3   
5           5            6         0       3   
6           6            7         0       1   
7           7            8         0       3   
8           8            9         1       3   
9           9           10         1       2   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   
5                                   Moran, Mr. James    male  27.1      0   
6                            McCarthy, Mr. Timothy J    male  54.0      0   
7                     Palsson, Master. Gosta Leonard    male   2.0      3   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  
5      0            330877   8.4583   NaN        Q  
6      0             17463  51.8625   E46        S  
7      1            349909  21.0750   NaN        S  
8      2            347742  11.1333   NaN        S  
9      0            237736  30.0708   NaN        C

# Prepare dataset

We will apply the following transformations on the dataset.

1. Tokenize the names. For example, "Braund, Mr. Owen Harris" will become ["Braund", "Mr.", "Owen", "Harris"].
2. Extract any prefix in the ticket. For example ticket "STON/O2. 3101282" will become "STON/O2." and 3101282.

In [3]:
train_df.drop(columns='Unnamed: 0', inplace=True)

In [4]:
def preprocess(df):
    df = df.copy()
    
    def normalize_name(x):
        return " ".join([v.strip(",()[].\"'") for v in x.split(" ")])
    
    def ticket_number(x):
        return x.split(" ")[-1]
        
    def ticket_item(x):
        items = x.split(" ")
        if len(items) == 1:
            return "NONE"
        return "_".join(items[0:-1])
    
    df["Name"] = df["Name"].apply(normalize_name)
    df["Ticket_number"] = df["Ticket"].apply(ticket_number)
    df["Ticket_item"] = df["Ticket"].apply(ticket_item)                     
    return df
    
preprocessed_train_df = preprocess(train_df)
preprocessed_serving_df = preprocess(serving_df)

preprocessed_train_df.head(5)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                              Name     Sex   Age  SibSp  \
0                            Braund Mr Owen Harris    male  22.0      1   
1  Cumings Mrs John Bradley Florence Briggs Thayer  female  38.0      1   
2                             Heikkinen Miss Laina  female  26.0      0   
3         Futrelle Mrs Jacques Heath Lily May Peel  female  35.0      1   
4                           Allen Mr William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked Ticket_number Ticket_item  
0      0         A/5 21171   7.2500   NaN        S         21171         A/5  
1      0          PC 17599  71.2833   C85        C         17599          PC  
2      0  STON/O2. 3101282   7.9250   NaN        S       3101282    STON/O2.  
3      0            113803  53.1000  C123        S        113803        NONE  
4      0            373450   8.0500   NaN        S        373450        NONE

Let's keep the list of the input features of the model. Notably, we don't want to train our model on the "PassengerId" and "Ticket" features.

In [5]:
input_features = list(preprocessed_train_df.columns)
input_features.remove("Ticket")
input_features.remove("PassengerId")
input_features.remove("Survived")
#input_features.remove("Ticket_number")

print(f"Input features: {input_features}")

Input features: ['Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Fare', 'Cabin', 'Embarked', 'Ticket_number', 'Ticket_item']


# Convert Pandas dataset to TensorFlow Dataset

In [6]:
def tokenize_names(features, labels=None):
    """Divite the names into tokens. TF-DF can consume text tokens natively."""
    features["Name"] =  tf.strings.split(features["Name"])
    return features, labels

train_ds = tfdf.keras.pd_dataframe_to_tf_dataset(preprocessed_train_df,label="Survived").map(tokenize_names)
serving_ds = tfdf.keras.pd_dataframe_to_tf_dataset(preprocessed_serving_df).map(tokenize_names)

2023-11-14 11:29:07.698256: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:981] could not open file to read NUMA node: /sys/bus/pci/devices/0000:01:00.0/numa_node
Your kernel may have been built without NUMA support.
2023-11-14 11:29:07.699976: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1960] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...


# Train model with default parameters

### Train model

First, we are training a GradientBoostedTreesModel model with the default parameters.

In [7]:
model = tfdf.keras.GradientBoostedTreesModel(
    verbose=0, # Very few logs
    features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
    exclude_non_specified_features=True, # Only use the features in "features"
    random_seed=1234,
)
model.fit(train_ds)

self_evaluation = model.make_inspector().evaluation()
print(f"Accuracy: {self_evaluation.accuracy} Loss:{self_evaluation.loss}")

[WARNING 23-11-14 11:29:08.2625 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:29:08.2627 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:29:08.2627 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:29:16.3922 NZDT kernel.cc:1243] Loading model from path /tmp/tmpd_29p7o0/model/ with prefix 1832264059184e66
[INFO 23-11-14 11:29:16.4247 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:29:16.4248 NZDT kernel.cc:1075] Use fast generic engine


Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: could not get source code
To silence this warning, decorate the function with @tf.autograph.experimental.do_not_convert


Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: could not get source code
To silence this warning, decorate the function with @tf.autograph.experimental.do_not_convert


Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: could not get source code
To silence this warning, decorate the function with @tf.autograph.experimental.do_not_convert
Accuracy: 0.804347813129425 Loss:0.8644342422485352


# Train model with improved default parameters

Now you'll use some specific parameters when creating the GBT model

In [8]:
model = tfdf.keras.GradientBoostedTreesModel(
    verbose=0, # Very few logs
    features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
    exclude_non_specified_features=True, # Only use the features in "features"
    
    #num_trees=2000,
    
    # Only for GBT.
    # A bit slower, but great to understand the model.
    # compute_permutation_variable_importance=True,
    
    # Change the default hyper-parameters
    # hyperparameter_template="benchmark_rank1@v1",
    
    #num_trees=1000,
    #tuner=tuner
    
    min_examples=1,
    categorical_algorithm="RANDOM",
    #max_depth=4,
    shrinkage=0.05,
    #num_candidate_attributes_ratio=0.2,
    split_axis="SPARSE_OBLIQUE",
    sparse_oblique_normalization="MIN_MAX",
    sparse_oblique_num_projections_exponent=2.0,
    num_trees=2000,
    #validation_ratio=0.0,
    random_seed=1234,
    
)
model.fit(train_ds)

self_evaluation = model.make_inspector().evaluation()
print(f"Accuracy: {self_evaluation.accuracy} Loss:{self_evaluation.loss}")

[WARNING 23-11-14 11:29:21.3739 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:29:21.3741 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:29:21.3742 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".


[INFO 23-11-14 11:29:24.4410 NZDT kernel.cc:1243] Loading model from path /tmp/tmpexdwf8zr/model/ with prefix 3222f33976104bb4
[INFO 23-11-14 11:29:24.4603 NZDT decision_forest.cc:660] Model loaded with 39 root(s), 2121 node(s), and 10 input feature(s).
[INFO 23-11-14 11:29:24.4604 NZDT kernel.cc:1075] Use fast generic engine


Accuracy: 0.79347825050354 Loss:1.015981674194336


Let's look at the model and you can also notice the information about variable importance that the model figured out

In [9]:
model.summary()

Model: "gradient_boosted_trees_model_1"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
Total params: 1 (1.00 Byte)
Trainable params: 0 (0.00 Byte)
Non-trainable params: 1 (1.00 Byte)
_________________________________________________________________
Type: "GRADIENT_BOOSTED_TREES"
Task: CLASSIFICATION
Label: "__LABEL"

Input Features (11):
	Age
	Cabin
	Embarked
	Fare
	Name
	Parch
	Pclass
	Sex
	SibSp
	Ticket_item
	Ticket_number

No weights

Variable Importance: INV_MEAN_MIN_DEPTH:
    1.           "Sex"  0.617983 ################
    2.           "Age"  0.394287 #######
    3.          "Fare"  0.261256 ###
    4.          "Name"  0.197417 
    5. "Ticket_number"  0.182452 
    6.   "Ticket_item"  0.176928 
    7.        "Pclass"  0.173694 
    8.         "Parch"  0.172633 
    9.      "Embarked"  0.172467 
   10.         "SibSp"  0.171490 

Variable Importance: NUM_AS_ROOT:
    1.  "Sex" 34.000000 #######

# Make predictions

In [10]:
def prediction_to_kaggle_format(model, threshold=0.5):
    proba_survive = model.predict(serving_ds, verbose=0)[:,0]
    return pd.DataFrame({
        "PassengerId": serving_df["PassengerId"],
        "Survived": (proba_survive >= threshold).astype(int)
    })

def make_submission(kaggle_predictions):
    try:
        path="/kaggle/working/submission1.csv"
        kaggle_predictions.to_csv(path, index=False)
        print(f"Submission exported to {path}")
    except:
        path="Dataset/submission1.csv"
        kaggle_predictions.to_csv(path, index=False)
        print(f"Submission exported to {path}")
kaggle_predictions = prediction_to_kaggle_format(model)
make_submission(kaggle_predictions)
try:
    !head /kaggle/working/submission1.csv
except:
    !head Dataset/submission1.csv


Submission exported to Dataset/submission1.csv
head: cannot open '/kaggle/working/submission1.csv' for reading: No such file or directory


# Training a model with hyperparameter tunning

Hyper-parameter tuning is enabled by specifying the tuner constructor argument of the model. The tuner object contains all the configuration of the tuner (search space, optimizer, trial and objective).


In [11]:
tuner = tfdf.tuner.RandomSearch(num_trials=1000)
tuner.choice("min_examples", [2, 5, 7, 10])
tuner.choice("categorical_algorithm", ["CART", "RANDOM"])

local_search_space = tuner.choice("growing_strategy", ["LOCAL"])
local_search_space.choice("max_depth", [3, 4, 5, 6, 8])

global_search_space = tuner.choice("growing_strategy", ["BEST_FIRST_GLOBAL"], merge=True)
global_search_space.choice("max_num_nodes", [16, 32, 64, 128, 256])

#tuner.choice("use_hessian_gain", [True, False])
tuner.choice("shrinkage", [0.02, 0.05, 0.10, 0.15])
tuner.choice("num_candidate_attributes_ratio", [0.2, 0.5, 0.9, 1.0])


tuner.choice("split_axis", ["AXIS_ALIGNED"])
oblique_space = tuner.choice("split_axis", ["SPARSE_OBLIQUE"], merge=True)
oblique_space.choice("sparse_oblique_normalization",
                     ["NONE", "STANDARD_DEVIATION", "MIN_MAX"])
oblique_space.choice("sparse_oblique_weights", ["BINARY", "CONTINUOUS"])
oblique_space.choice("sparse_oblique_num_projections_exponent", [1.0, 1.5])

# Tune the model. Notice the `tuner=tuner`.
tuned_model = tfdf.keras.GradientBoostedTreesModel(tuner=tuner)
tuned_model.fit(train_ds, verbose=0)

tuned_self_evaluation = tuned_model.make_inspector().evaluation()
print(f"Accuracy: {tuned_self_evaluation.accuracy} Loss:{tuned_self_evaluation.loss}")

Use /tmp/tmpk8vklnha as temporary training directory


[WARNING 23-11-14 11:29:26.7306 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:29:26.7307 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:29:26.7308 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:30:44.7549 NZDT kernel.cc:1243] Loading model from path /tmp/tmpk8vklnha/model/ with prefix 0b5e2ec3111f4903
[INFO 23-11-14 11:30:44.7735 NZDT decision_forest.cc:660] Model loaded with 20 root(s), 1696 node(s), and 12 input feature(s).
[INFO 23-11-14 11:30:44.7735 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesGeneric" built
[INFO 23-11-14 11:30:44.7735 NZDT kernel.cc:1075] Use fast generic engine


Accuracy: 0.8904109597206116 Loss:0.630903959274292


In the last line in the cell above, you can see the accuracy is higher than previously with default parameters and parameters set by hand.

This is the main idea behing hyperparameter tuning.

For more information you can follow this tutorial: [Automated hyper-parameter tuning](https://www.tensorflow.org/decision_forests/tutorials/automatic_tuning_colab)

# Making an ensemble

Here you'll create 100 models with different seeds and combine their results

This approach removes a little bit the random aspects related to creating ML models

In the GBT creation is used the `honest` parameter. It will use different training examples to infer the structure and the leaf values. This regularization technique trades examples for bias estimates.

In [12]:
predictions = None
num_predictions = 0

for i in range(100):
    print(f"i:{i}")
    # Possible models: GradientBoostedTreesModel or RandomForestModel
    model = tfdf.keras.GradientBoostedTreesModel(
        verbose=0, # Very few logs
        features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
        exclude_non_specified_features=True, # Only use the features in "features"

        #min_examples=1,
        #categorical_algorithm="RANDOM",
        ##max_depth=4,
        #shrinkage=0.05,
        ##num_candidate_attributes_ratio=0.2,
        #split_axis="SPARSE_OBLIQUE",
        #sparse_oblique_normalization="MIN_MAX",
        #sparse_oblique_num_projections_exponent=2.0,
        #num_trees=2000,
        ##validation_ratio=0.0,
        random_seed=i,
        honest=True,
    )
    model.fit(train_ds)
    
    sub_predictions = model.predict(serving_ds, verbose=0)[:,0]
    if predictions is None:
        predictions = sub_predictions
    else:
        predictions += sub_predictions
    num_predictions += 1

predictions/=num_predictions

kaggle_predictions = pd.DataFrame({
        "PassengerId": serving_df["PassengerId"],
        "Survived": (predictions >= 0.5).astype(int)
    })

make_submission(kaggle_predictions)

i:0


[WARNING 23-11-14 11:30:45.4363 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:30:45.4363 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:30:45.4364 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:30:47.0715 NZDT kernel.cc:1243] Loading model from path /tmp/tmp2m4166ss/model/ with prefix f5adf432cf014bc8
[INFO 23-11-14 11:30:47.0812 NZDT kernel.cc:1075] Use fast generic engine


i:1


[WARNING 23-11-14 11:30:48.2200 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:30:48.2201 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:30:48.2202 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".


[INFO 23-11-14 11:30:50.1737 NZDT kernel.cc:1243] Loading model from path /tmp/tmpl3g9tojt/model/ with prefix 96f7dfa44df54679
[INFO 23-11-14 11:30:50.1910 NZDT kernel.cc:1075] Use fast generic engine


i:2


[WARNING 23-11-14 11:30:51.3939 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:30:51.3939 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:30:51.3940 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".


[INFO 23-11-14 11:30:55.1961 NZDT kernel.cc:1243] Loading model from path /tmp/tmps95xjpni/model/ with prefix cd0c0570acc142be
[INFO 23-11-14 11:30:55.2215 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:30:55.2216 NZDT kernel.cc:1075] Use fast generic engine


i:3


[WARNING 23-11-14 11:30:55.9240 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:30:55.9241 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:30:55.9241 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:30:57.4669 NZDT kernel.cc:1243] Loading model from path /tmp/tmpmpmib42p/model/ with prefix a5a302dc5d9e49b6
[INFO 23-11-14 11:30:57.4789 NZDT kernel.cc:1075] Use fast generic engine


i:4


[WARNING 23-11-14 11:30:58.2352 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:30:58.2352 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:30:58.2353 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:00.7431 NZDT kernel.cc:1243] Loading model from path /tmp/tmps46sh40b/model/ with prefix f9274b3b51744073
[INFO 23-11-14 11:31:00.7754 NZDT kernel.cc:1075] Use fast generic engine


i:5


[WARNING 23-11-14 11:31:02.1398 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:02.1399 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:02.1399 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:04.3080 NZDT kernel.cc:1243] Loading model from path /tmp/tmplcs3_7qn/model/ with prefix aab44d41d58d4f5e
[INFO 23-11-14 11:31:04.3257 NZDT kernel.cc:1075] Use fast generic engine


i:6


[WARNING 23-11-14 11:31:05.0243 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:05.0243 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:05.0243 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:06.7663 NZDT kernel.cc:1243] Loading model from path /tmp/tmp7xrqxa4_/model/ with prefix 45978244420a4258
[INFO 23-11-14 11:31:06.7762 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:31:06.7762 NZDT kernel.cc:1075] Use fast generic engine


i:7


[WARNING 23-11-14 11:31:07.4876 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:07.4876 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:07.4877 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:09.0485 NZDT kernel.cc:1243] Loading model from path /tmp/tmp2dbe_rza/model/ with prefix 2643627e3c784ed3
[INFO 23-11-14 11:31:09.0569 NZDT kernel.cc:1075] Use fast generic engine


i:8


[WARNING 23-11-14 11:31:10.2776 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:10.2776 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:10.2776 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:12.4749 NZDT kernel.cc:1243] Loading model from path /tmp/tmp_85jrjvf/model/ with prefix 6703e56bb35a4d9d
[INFO 23-11-14 11:31:12.4925 NZDT kernel.cc:1075] Use fast generic engine


i:9


[WARNING 23-11-14 11:31:13.6272 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:13.6273 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:13.6273 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:15.4854 NZDT kernel.cc:1243] Loading model from path /tmp/tmp8p38puw4/model/ with prefix ce98b076990b4eb5
[INFO 23-11-14 11:31:15.5026 NZDT kernel.cc:1075] Use fast generic engine


i:10


[WARNING 23-11-14 11:31:16.3572 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:16.3572 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:16.3573 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:17.9201 NZDT kernel.cc:1243] Loading model from path /tmp/tmpp4xbtou_/model/ with prefix c07237efd1a74562
[INFO 23-11-14 11:31:17.9289 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:31:17.9290 NZDT kernel.cc:1075] Use fast generic engine


i:11


[WARNING 23-11-14 11:31:18.6865 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:18.6865 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:18.6865 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:20.1899 NZDT kernel.cc:1243] Loading model from path /tmp/tmpn6lbiy9q/model/ with prefix 6fd5fdb9b78b4577
[INFO 23-11-14 11:31:20.1990 NZDT kernel.cc:1075] Use fast generic engine


i:12


[WARNING 23-11-14 11:31:21.2627 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:21.2627 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:21.2628 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:23.2527 NZDT kernel.cc:1243] Loading model from path /tmp/tmpd0wua_yh/model/ with prefix 5bce19e1ff74441f
[INFO 23-11-14 11:31:23.2669 NZDT kernel.cc:1075] Use fast generic engine


i:13


[WARNING 23-11-14 11:31:25.2548 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:25.2549 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:25.2549 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:27.5849 NZDT kernel.cc:1243] Loading model from path /tmp/tmpnsfv0dam/model/ with prefix c9c6a1034e8f4e3e
[INFO 23-11-14 11:31:27.6021 NZDT kernel.cc:1075] Use fast generic engine


i:14


[WARNING 23-11-14 11:31:28.4758 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:28.4759 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:28.4759 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:30.7915 NZDT kernel.cc:1243] Loading model from path /tmp/tmpe87wchvk/model/ with prefix 57f5b13781d94007
[INFO 23-11-14 11:31:30.8177 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:31:30.8178 NZDT kernel.cc:1075] Use fast generic engine


i:15


[WARNING 23-11-14 11:31:31.7606 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:31.7606 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:31.7607 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:33.0692 NZDT kernel.cc:1243] Loading model from path /tmp/tmp9l5701a5/model/ with prefix c21d67b36c904bed
[INFO 23-11-14 11:31:33.0737 NZDT kernel.cc:1075] Use fast generic engine


i:16


[WARNING 23-11-14 11:31:34.0007 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:34.0007 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:34.0007 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:35.7068 NZDT kernel.cc:1243] Loading model from path /tmp/tmp9fxwg4g1/model/ with prefix ebbaa31201fc4aa2
[INFO 23-11-14 11:31:35.7240 NZDT kernel.cc:1075] Use fast generic engine


i:17


[WARNING 23-11-14 11:31:36.7955 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:36.7956 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:36.7956 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:38.9821 NZDT kernel.cc:1243] Loading model from path /tmp/tmptospikzr/model/ with prefix f84e306507f8440d
[INFO 23-11-14 11:31:39.0011 NZDT kernel.cc:1075] Use fast generic engine


i:18


[WARNING 23-11-14 11:31:39.8717 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:39.8718 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:39.8718 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:41.9596 NZDT kernel.cc:1243] Loading model from path /tmp/tmpn47worwb/model/ with prefix 3ec4b97a6aa642fc
[INFO 23-11-14 11:31:42.0023 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:31:42.0024 NZDT kernel.cc:1075] Use fast generic engine


i:19


[WARNING 23-11-14 11:31:43.0083 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:43.0084 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:43.0084 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:45.1598 NZDT kernel.cc:1243] Loading model from path /tmp/tmph3bifh8c/model/ with prefix dc088eb7c8dd4ca4
[INFO 23-11-14 11:31:45.1765 NZDT kernel.cc:1075] Use fast generic engine


i:20


[WARNING 23-11-14 11:31:45.9132 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:45.9133 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:45.9133 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:48.8660 NZDT kernel.cc:1243] Loading model from path /tmp/tmpfjrdgc6s/model/ with prefix 63e7ca0d678542b8
[INFO 23-11-14 11:31:48.9007 NZDT kernel.cc:1075] Use fast generic engine


i:21


[WARNING 23-11-14 11:31:49.7576 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:49.7577 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:49.7577 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:51.4073 NZDT kernel.cc:1243] Loading model from path /tmp/tmptvb7lbmx/model/ with prefix 11f69059760f428b
[INFO 23-11-14 11:31:51.4195 NZDT kernel.cc:1075] Use fast generic engine


i:22


[WARNING 23-11-14 11:31:52.9865 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:52.9865 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:52.9866 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:54.8417 NZDT kernel.cc:1243] Loading model from path /tmp/tmp0dow4vv2/model/ with prefix ca06266b2e0646ac
[INFO 23-11-14 11:31:54.8496 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:31:54.8497 NZDT kernel.cc:1075] Use fast generic engine


i:23


[WARNING 23-11-14 11:31:56.0487 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:56.0487 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:56.0488 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:31:58.9850 NZDT kernel.cc:1243] Loading model from path /tmp/tmpdh_gp3zl/model/ with prefix 9f66a4020bb9464b
[INFO 23-11-14 11:31:59.0015 NZDT kernel.cc:1075] Use fast generic engine


i:24


[WARNING 23-11-14 11:31:59.7198 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:59.7198 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:31:59.7198 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:32:01.3675 NZDT kernel.cc:1243] Loading model from path /tmp/tmpl6jrq2kx/model/ with prefix b449adb088c648e1
[INFO 23-11-14 11:32:01.3760 NZDT kernel.cc:1075] Use fast generic engine


i:25


[WARNING 23-11-14 11:32:02.2046 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:02.2047 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:02.2047 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:32:04.2799 NZDT kernel.cc:1243] Loading model from path /tmp/tmp_nf8jtke/model/ with prefix 8b4c7b7965a94a0a
[INFO 23-11-14 11:32:04.3082 NZDT kernel.cc:1075] Use fast generic engine


i:26


[WARNING 23-11-14 11:32:05.0085 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:05.0086 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:05.0086 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:32:06.8200 NZDT kernel.cc:1243] Loading model from path /tmp/tmp1cwcl7h7/model/ with prefix d594cf6499564769
[INFO 23-11-14 11:32:06.8325 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:32:06.8325 NZDT kernel.cc:1075] Use fast generic engine


i:27


[WARNING 23-11-14 11:32:07.6974 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:07.6974 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:07.6974 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:32:09.5874 NZDT kernel.cc:1243] Loading model from path /tmp/tmpe58ftu3v/model/ with prefix 7b70209843d24236
[INFO 23-11-14 11:32:09.5968 NZDT kernel.cc:1075] Use fast generic engine


i:28


[WARNING 23-11-14 11:32:10.6809 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:10.6809 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:10.6809 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:32:12.3805 NZDT kernel.cc:1243] Loading model from path /tmp/tmpr0ovdjrd/model/ with prefix bf986ccf209c437a
[INFO 23-11-14 11:32:12.3915 NZDT kernel.cc:1075] Use fast generic engine


i:29


[WARNING 23-11-14 11:32:13.4649 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:13.4650 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:13.4651 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:32:15.4517 NZDT kernel.cc:1243] Loading model from path /tmp/tmp5pn1eett/model/ with prefix bb8bbc1eaef245d8
[INFO 23-11-14 11:32:15.4610 NZDT kernel.cc:1075] Use fast generic engine


i:30


[WARNING 23-11-14 11:32:16.5414 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:16.5415 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:16.5415 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:32:19.0717 NZDT kernel.cc:1243] Loading model from path /tmp/tmpr1kjg_bl/model/ with prefix 425119bcba0c4834
[INFO 23-11-14 11:32:19.0818 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:32:19.0818 NZDT kernel.cc:1075] Use fast generic engine


i:31


[WARNING 23-11-14 11:32:20.0160 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:20.0161 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:20.0161 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:32:22.3654 NZDT kernel.cc:1243] Loading model from path /tmp/tmppsmg5fos/model/ with prefix 92f6eeff2403426d
[INFO 23-11-14 11:32:22.3760 NZDT kernel.cc:1075] Use fast generic engine


i:32


[WARNING 23-11-14 11:32:23.2989 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:23.2989 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:23.2990 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:32:25.2841 NZDT kernel.cc:1243] Loading model from path /tmp/tmpzziij51m/model/ with prefix 96341f8f07fd40f1
[INFO 23-11-14 11:32:25.2939 NZDT kernel.cc:1075] Use fast generic engine


i:33


[WARNING 23-11-14 11:32:26.4145 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:26.4146 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:26.4146 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:32:28.4107 NZDT kernel.cc:1243] Loading model from path /tmp/tmp4oz_rdb3/model/ with prefix 496b671b565046b9
[INFO 23-11-14 11:32:28.4220 NZDT kernel.cc:1075] Use fast generic engine


i:34


[WARNING 23-11-14 11:32:29.3232 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:29.3232 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:29.3233 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:32:31.3481 NZDT kernel.cc:1243] Loading model from path /tmp/tmphpml0fol/model/ with prefix ef6935de61d440ea
[INFO 23-11-14 11:32:31.3557 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:32:31.3557 NZDT kernel.cc:1075] Use fast generic engine


i:35


[WARNING 23-11-14 11:32:32.3283 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:32.3284 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:32.3284 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:32:34.8818 NZDT kernel.cc:1243] Loading model from path /tmp/tmp4j9h6lmo/model/ with prefix 3cc55134e9714095
[INFO 23-11-14 11:32:34.8915 NZDT kernel.cc:1075] Use fast generic engine


i:36


[WARNING 23-11-14 11:32:36.4448 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:36.4448 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:36.4449 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:32:38.5947 NZDT kernel.cc:1243] Loading model from path /tmp/tmpytashc6r/model/ with prefix da65d16a019f42e8
[INFO 23-11-14 11:32:38.6013 NZDT kernel.cc:1075] Use fast generic engine


i:37


[WARNING 23-11-14 11:32:40.1222 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:40.1223 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:40.1224 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:32:41.8554 NZDT kernel.cc:1243] Loading model from path /tmp/tmpke788902/model/ with prefix 95e1d25e62374622
[INFO 23-11-14 11:32:41.8611 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:32:41.8612 NZDT kernel.cc:1075] Use fast generic engine


i:38


[WARNING 23-11-14 11:32:43.3741 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:43.3741 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:43.3742 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:32:46.2245 NZDT kernel.cc:1243] Loading model from path /tmp/tmpwv6sc1_q/model/ with prefix a92d7fd82f3c4656
[INFO 23-11-14 11:32:46.2398 NZDT kernel.cc:1075] Use fast generic engine


i:39


[WARNING 23-11-14 11:32:48.6524 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:48.6525 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:48.6526 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:32:51.1264 NZDT kernel.cc:1243] Loading model from path /tmp/tmpyi01_p1m/model/ with prefix 546e0215f128480b
[INFO 23-11-14 11:32:51.1357 NZDT kernel.cc:1075] Use fast generic engine


i:40


[WARNING 23-11-14 11:32:52.3204 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:52.3205 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:52.3205 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:32:54.0935 NZDT kernel.cc:1243] Loading model from path /tmp/tmp535q5hxh/model/ with prefix 5e070e98394044a2
[INFO 23-11-14 11:32:54.1003 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:32:54.1003 NZDT kernel.cc:1075] Use fast generic engine


i:41


[WARNING 23-11-14 11:32:56.0374 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:56.0375 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:56.0375 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:32:58.6069 NZDT kernel.cc:1243] Loading model from path /tmp/tmp7pldnh4v/model/ with prefix ac643550e17c4f81
[INFO 23-11-14 11:32:58.6153 NZDT kernel.cc:1075] Use fast generic engine


i:42


[WARNING 23-11-14 11:32:59.9734 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:59.9735 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:32:59.9735 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:33:02.1404 NZDT kernel.cc:1243] Loading model from path /tmp/tmpcgz39c2o/model/ with prefix 27b54b65089843fa
[INFO 23-11-14 11:33:02.1532 NZDT kernel.cc:1075] Use fast generic engine


i:43


[WARNING 23-11-14 11:33:03.8827 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:03.8828 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:03.8828 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:33:06.1829 NZDT kernel.cc:1243] Loading model from path /tmp/tmp9hvt81ft/model/ with prefix 16566b6e71e94a59
[INFO 23-11-14 11:33:06.1934 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:33:06.1935 NZDT kernel.cc:1075] Use fast generic engine


i:44


[WARNING 23-11-14 11:33:07.2860 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:07.2860 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:07.2860 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:33:09.0162 NZDT kernel.cc:1243] Loading model from path /tmp/tmpu3n0_pp1/model/ with prefix f2fc392a7b934827
[INFO 23-11-14 11:33:09.0187 NZDT kernel.cc:1075] Use fast generic engine


i:45


[WARNING 23-11-14 11:33:10.6605 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:10.6605 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:10.6605 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:33:12.1722 NZDT kernel.cc:1243] Loading model from path /tmp/tmpaglez3lg/model/ with prefix 8bc1cdbae4ef43dd
[INFO 23-11-14 11:33:12.1775 NZDT kernel.cc:1075] Use fast generic engine


i:46


[WARNING 23-11-14 11:33:13.5989 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:13.5990 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:13.5990 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:33:15.8252 NZDT kernel.cc:1243] Loading model from path /tmp/tmpfy_8o0c7/model/ with prefix c6457a7e0b7d4813
[INFO 23-11-14 11:33:15.8327 NZDT kernel.cc:1075] Use fast generic engine


i:47


[WARNING 23-11-14 11:33:16.8473 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:16.8475 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:16.8476 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:33:18.5944 NZDT kernel.cc:1243] Loading model from path /tmp/tmpo5ml3vqg/model/ with prefix e033b40f0f294155
[INFO 23-11-14 11:33:18.6057 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:33:18.6057 NZDT kernel.cc:1075] Use fast generic engine


i:48


[WARNING 23-11-14 11:33:19.8776 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:19.8777 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:19.8777 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:33:21.3534 NZDT kernel.cc:1243] Loading model from path /tmp/tmpu_zmpepr/model/ with prefix faf53b86d64b4bdf
[INFO 23-11-14 11:33:21.3587 NZDT kernel.cc:1075] Use fast generic engine


i:49


[WARNING 23-11-14 11:33:22.6918 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:22.6918 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:22.6919 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:33:24.7688 NZDT kernel.cc:1243] Loading model from path /tmp/tmpr3gde4_a/model/ with prefix aae0b08c432a40dc
[INFO 23-11-14 11:33:24.7759 NZDT kernel.cc:1075] Use fast generic engine


i:50


[WARNING 23-11-14 11:33:26.2688 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:26.2689 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:26.2689 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:33:29.7116 NZDT kernel.cc:1243] Loading model from path /tmp/tmpv4rvuzbd/model/ with prefix 1d3b7b4bf4f04dd4
[INFO 23-11-14 11:33:29.7499 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:33:29.7499 NZDT kernel.cc:1075] Use fast generic engine


i:51


[WARNING 23-11-14 11:33:31.2160 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:31.2160 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:31.2160 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:33:33.8826 NZDT kernel.cc:1243] Loading model from path /tmp/tmp4moe_v9t/model/ with prefix 0fb4c2ba028a48c2
[INFO 23-11-14 11:33:33.8954 NZDT kernel.cc:1075] Use fast generic engine


i:52


[WARNING 23-11-14 11:33:35.5485 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:35.5486 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:35.5486 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:33:38.0116 NZDT kernel.cc:1243] Loading model from path /tmp/tmpnpxonlwj/model/ with prefix 7417b11b96124ad3
[INFO 23-11-14 11:33:38.0216 NZDT kernel.cc:1075] Use fast generic engine


i:53


[WARNING 23-11-14 11:33:39.1671 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:39.1672 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:39.1673 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:33:41.9201 NZDT kernel.cc:1243] Loading model from path /tmp/tmp9dnvymo9/model/ with prefix 84a8805742784be0
[INFO 23-11-14 11:33:41.9577 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:33:41.9578 NZDT kernel.cc:1075] Use fast generic engine


i:54


[WARNING 23-11-14 11:33:42.9689 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:42.9691 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:42.9692 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:33:45.2533 NZDT kernel.cc:1243] Loading model from path /tmp/tmp18al4er2/model/ with prefix cf13faad2a8f47aa
[INFO 23-11-14 11:33:45.2636 NZDT kernel.cc:1075] Use fast generic engine


i:55


[WARNING 23-11-14 11:33:47.7112 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:47.7113 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:47.7113 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:33:50.5175 NZDT kernel.cc:1243] Loading model from path /tmp/tmp65620k0n/model/ with prefix 10ac6c5962b74949
[INFO 23-11-14 11:33:50.5289 NZDT kernel.cc:1075] Use fast generic engine


i:56


[WARNING 23-11-14 11:33:52.6279 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:52.6280 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:52.6280 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:33:55.4208 NZDT kernel.cc:1243] Loading model from path /tmp/tmp59la14qv/model/ with prefix 980a66ff0a4d404d
[INFO 23-11-14 11:33:55.4545 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:33:55.4546 NZDT kernel.cc:1075] Use fast generic engine


i:57


[WARNING 23-11-14 11:33:56.3872 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:56.3872 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:33:56.3872 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:33:58.0230 NZDT kernel.cc:1243] Loading model from path /tmp/tmpl252p3un/model/ with prefix a256a8139bc948d5
[INFO 23-11-14 11:33:58.0343 NZDT kernel.cc:1075] Use fast generic engine


i:58


[WARNING 23-11-14 11:34:00.6272 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:00.6273 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:00.6273 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:34:03.1613 NZDT kernel.cc:1243] Loading model from path /tmp/tmpitu0xqqv/model/ with prefix 0ae509c9fa9f444b
[INFO 23-11-14 11:34:03.1744 NZDT kernel.cc:1075] Use fast generic engine


i:59


[WARNING 23-11-14 11:34:04.6206 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:04.6206 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:04.6207 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:34:08.3790 NZDT kernel.cc:1243] Loading model from path /tmp/tmpt_zobfp9/model/ with prefix 0488d7627ced4659
[INFO 23-11-14 11:34:08.4305 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:34:08.4306 NZDT kernel.cc:1075] Use fast generic engine


i:60


[WARNING 23-11-14 11:34:10.4470 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:10.4470 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:10.4471 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:34:13.4438 NZDT kernel.cc:1243] Loading model from path /tmp/tmpm_fjw_14/model/ with prefix 3327710f0a8d4ee9
[INFO 23-11-14 11:34:13.4596 NZDT kernel.cc:1075] Use fast generic engine


i:61


[WARNING 23-11-14 11:34:14.7357 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:14.7358 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:14.7358 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:34:16.8589 NZDT kernel.cc:1243] Loading model from path /tmp/tmp6llcbpr9/model/ with prefix 923b1a0ed4b14eb1
[INFO 23-11-14 11:34:16.8672 NZDT kernel.cc:1075] Use fast generic engine


i:62


[WARNING 23-11-14 11:34:17.7937 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:17.7938 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:17.7938 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:34:19.7928 NZDT kernel.cc:1243] Loading model from path /tmp/tmpre3j0nnb/model/ with prefix fa410d49d5b4441a
[INFO 23-11-14 11:34:19.8010 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:34:19.8010 NZDT kernel.cc:1075] Use fast generic engine


i:63


[WARNING 23-11-14 11:34:21.3129 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:21.3130 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:21.3130 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:34:23.5266 NZDT kernel.cc:1243] Loading model from path /tmp/tmp1yd7hane/model/ with prefix 02d6224c8c0d414e
[INFO 23-11-14 11:34:23.5423 NZDT kernel.cc:1075] Use fast generic engine


i:64


[WARNING 23-11-14 11:34:24.8742 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:24.8742 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:24.8743 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:34:27.2855 NZDT kernel.cc:1243] Loading model from path /tmp/tmp5tphrnos/model/ with prefix a7a5fc2cb7cf4acd
[INFO 23-11-14 11:34:27.2962 NZDT kernel.cc:1075] Use fast generic engine


i:65


[WARNING 23-11-14 11:34:28.4254 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:28.4255 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:28.4255 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:34:30.1916 NZDT kernel.cc:1243] Loading model from path /tmp/tmpzvwf88sr/model/ with prefix 836c6a300a9c43c3
[INFO 23-11-14 11:34:30.1991 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:34:30.1992 NZDT kernel.cc:1075] Use fast generic engine


i:66


[WARNING 23-11-14 11:34:32.0453 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:32.0454 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:32.0454 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:34:35.5919 NZDT kernel.cc:1243] Loading model from path /tmp/tmpg3r5wdp5/model/ with prefix 89c8c8f280ee4aae
[INFO 23-11-14 11:34:35.6128 NZDT kernel.cc:1075] Use fast generic engine


i:67


[WARNING 23-11-14 11:34:36.6591 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:36.6592 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:36.6592 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:34:40.0158 NZDT kernel.cc:1243] Loading model from path /tmp/tmp_h1e4j_p/model/ with prefix f659c1a51f724c29
[INFO 23-11-14 11:34:40.0810 NZDT kernel.cc:1075] Use fast generic engine


i:68


[WARNING 23-11-14 11:34:41.4645 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:41.4645 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:41.4646 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:34:43.4310 NZDT kernel.cc:1243] Loading model from path /tmp/tmp_ceb7o2c/model/ with prefix 4618dfd8baa349d2
[INFO 23-11-14 11:34:43.4442 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:34:43.4443 NZDT kernel.cc:1075] Use fast generic engine


i:69


[WARNING 23-11-14 11:34:44.7536 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:44.7536 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:44.7536 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:34:46.7770 NZDT kernel.cc:1243] Loading model from path /tmp/tmpipltoj8u/model/ with prefix 81e64415a42d488f
[INFO 23-11-14 11:34:46.7817 NZDT kernel.cc:1075] Use fast generic engine


i:70


[WARNING 23-11-14 11:34:47.6978 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:47.6978 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:47.6979 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:34:49.6746 NZDT kernel.cc:1243] Loading model from path /tmp/tmpo2cl2v8c/model/ with prefix 7f58dd9d44424f90
[INFO 23-11-14 11:34:49.6818 NZDT kernel.cc:1075] Use fast generic engine


i:71


[WARNING 23-11-14 11:34:51.1052 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:51.1054 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:51.1054 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:34:53.2113 NZDT kernel.cc:1243] Loading model from path /tmp/tmpghp18v4m/model/ with prefix ba7c63f0361e44b9
[INFO 23-11-14 11:34:53.2199 NZDT kernel.cc:1075] Use fast generic engine


i:72


[WARNING 23-11-14 11:34:54.8751 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:54.8751 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:54.8751 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:34:57.3479 NZDT kernel.cc:1243] Loading model from path /tmp/tmp_e3nipxq/model/ with prefix 19165324a2c34159
[INFO 23-11-14 11:34:57.3764 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:34:57.3765 NZDT kernel.cc:1075] Use fast generic engine


i:73


[WARNING 23-11-14 11:34:59.0986 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:59.0986 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:34:59.0986 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:35:00.7966 NZDT kernel.cc:1243] Loading model from path /tmp/tmpu9l9wx2a/model/ with prefix 24b6941d07fe41b7
[INFO 23-11-14 11:35:00.8053 NZDT kernel.cc:1075] Use fast generic engine


i:74


[WARNING 23-11-14 11:35:01.9341 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:01.9341 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:01.9342 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:35:07.4346 NZDT kernel.cc:1243] Loading model from path /tmp/tmps8sp9y89/model/ with prefix d9b59df00c2a4256
[INFO 23-11-14 11:35:07.5531 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:35:07.5532 NZDT kernel.cc:1075] Use fast generic engine


i:75


[WARNING 23-11-14 11:35:08.7526 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:08.7527 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:08.7527 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:35:12.9201 NZDT kernel.cc:1243] Loading model from path /tmp/tmpzoqqrkvg/model/ with prefix cb41371856464ece
[INFO 23-11-14 11:35:12.9710 NZDT kernel.cc:1075] Use fast generic engine


i:76


[WARNING 23-11-14 11:35:14.6667 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:14.6667 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:14.6668 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:35:16.8770 NZDT kernel.cc:1243] Loading model from path /tmp/tmpgykilzk_/model/ with prefix cf58c80ad9004054
[INFO 23-11-14 11:35:16.8849 NZDT kernel.cc:1075] Use fast generic engine


i:77


[WARNING 23-11-14 11:35:18.6396 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:18.6396 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:18.6396 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:35:20.9045 NZDT kernel.cc:1243] Loading model from path /tmp/tmp5niq0mj4/model/ with prefix e4c53d5455334da2
[INFO 23-11-14 11:35:20.9134 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:35:20.9135 NZDT kernel.cc:1075] Use fast generic engine


i:78


[WARNING 23-11-14 11:35:22.6317 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:22.6317 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:22.6317 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:35:24.2557 NZDT kernel.cc:1243] Loading model from path /tmp/tmpw9xcb29m/model/ with prefix 1d4868686f6a45aa
[INFO 23-11-14 11:35:24.2602 NZDT kernel.cc:1075] Use fast generic engine


i:79


[WARNING 23-11-14 11:35:25.5800 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:25.5801 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:25.5801 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:35:29.4807 NZDT kernel.cc:1243] Loading model from path /tmp/tmp44kccjx3/model/ with prefix c048d6c01bad4d45
[INFO 23-11-14 11:35:29.4948 NZDT kernel.cc:1075] Use fast generic engine


i:80


[WARNING 23-11-14 11:35:30.8148 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:30.8149 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:30.8149 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:35:32.9206 NZDT kernel.cc:1243] Loading model from path /tmp/tmpv8s9e564/model/ with prefix 96cd13cc065341fb
[INFO 23-11-14 11:35:32.9358 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:35:32.9359 NZDT kernel.cc:1075] Use fast generic engine


i:81


[WARNING 23-11-14 11:35:34.2407 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:34.2408 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:34.2408 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:35:36.2524 NZDT kernel.cc:1243] Loading model from path /tmp/tmpyg4m35pb/model/ with prefix 628f012284d140b9
[INFO 23-11-14 11:35:36.2625 NZDT kernel.cc:1075] Use fast generic engine


i:82


[WARNING 23-11-14 11:35:37.9183 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:37.9184 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:37.9184 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:35:40.3193 NZDT kernel.cc:1243] Loading model from path /tmp/tmp3k6k4_ce/model/ with prefix 2b25d533db4e4183
[INFO 23-11-14 11:35:40.3295 NZDT kernel.cc:1075] Use fast generic engine


i:83


[WARNING 23-11-14 11:35:41.6570 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:41.6571 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:41.6571 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:35:43.7981 NZDT kernel.cc:1243] Loading model from path /tmp/tmpiqln5epa/model/ with prefix b5d2e79321b74a5e
[INFO 23-11-14 11:35:43.8030 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:35:43.8030 NZDT kernel.cc:1075] Use fast generic engine


i:84


[WARNING 23-11-14 11:35:45.6716 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:45.6717 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:45.6717 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:35:47.8542 NZDT kernel.cc:1243] Loading model from path /tmp/tmpg_yodirg/model/ with prefix e9a8ace5cb58425a
[INFO 23-11-14 11:35:47.8610 NZDT kernel.cc:1075] Use fast generic engine


i:85


[WARNING 23-11-14 11:35:48.9004 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:48.9004 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:48.9005 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:35:50.8470 NZDT kernel.cc:1243] Loading model from path /tmp/tmp2pm5z3vh/model/ with prefix 94c6bf0ab5ab4aa9
[INFO 23-11-14 11:35:50.8561 NZDT kernel.cc:1075] Use fast generic engine


i:86


[WARNING 23-11-14 11:35:52.5720 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:52.5721 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:52.5721 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:35:57.5859 NZDT kernel.cc:1243] Loading model from path /tmp/tmpiefz3tdi/model/ with prefix 975590529fa548f5
[INFO 23-11-14 11:35:57.6443 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:35:57.6443 NZDT kernel.cc:1075] Use fast generic engine


i:87


[WARNING 23-11-14 11:35:59.5477 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:59.5478 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:35:59.5479 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:36:02.2992 NZDT kernel.cc:1243] Loading model from path /tmp/tmph4689eu_/model/ with prefix bbaa64b4fa5044a9
[INFO 23-11-14 11:36:02.3293 NZDT kernel.cc:1075] Use fast generic engine


i:88


[WARNING 23-11-14 11:36:03.5418 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:03.5419 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:03.5419 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:36:05.5489 NZDT kernel.cc:1243] Loading model from path /tmp/tmpvyuei6us/model/ with prefix 5b16102c98ae4fa2
[INFO 23-11-14 11:36:05.5564 NZDT kernel.cc:1075] Use fast generic engine


i:89


[WARNING 23-11-14 11:36:06.5647 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:06.5647 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:06.5648 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:36:09.2828 NZDT kernel.cc:1243] Loading model from path /tmp/tmp9_m845xw/model/ with prefix 554e4331f7954375
[INFO 23-11-14 11:36:09.2972 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:36:09.2973 NZDT kernel.cc:1075] Use fast generic engine


i:90


[WARNING 23-11-14 11:36:10.6953 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:10.6954 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:10.6954 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:36:13.3488 NZDT kernel.cc:1243] Loading model from path /tmp/tmpzsc7clni/model/ with prefix 8ccdddd2e5974e46
[INFO 23-11-14 11:36:13.3620 NZDT kernel.cc:1075] Use fast generic engine


i:91


[WARNING 23-11-14 11:36:14.6384 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:14.6384 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:14.6384 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:36:17.5910 NZDT kernel.cc:1243] Loading model from path /tmp/tmp2xvfwpys/model/ with prefix 194107db46ef4739
[INFO 23-11-14 11:36:17.6031 NZDT kernel.cc:1075] Use fast generic engine


i:92


[WARNING 23-11-14 11:36:18.7356 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:18.7357 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:18.7357 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:36:21.4617 NZDT kernel.cc:1243] Loading model from path /tmp/tmpmc4xlple/model/ with prefix 8d6ddf030a1c442d
[INFO 23-11-14 11:36:21.4991 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:36:21.4992 NZDT kernel.cc:1075] Use fast generic engine


i:93


[WARNING 23-11-14 11:36:23.3296 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:23.3296 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:23.3297 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:36:25.8245 NZDT kernel.cc:1243] Loading model from path /tmp/tmp97l4nu_w/model/ with prefix 917151129e7741c9
[INFO 23-11-14 11:36:25.8387 NZDT kernel.cc:1075] Use fast generic engine


i:94


[WARNING 23-11-14 11:36:26.8598 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:26.8598 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:26.8599 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:36:30.0765 NZDT kernel.cc:1243] Loading model from path /tmp/tmp3b2wnro_/model/ with prefix 7c3c1fbca7f94a54
[INFO 23-11-14 11:36:30.1244 NZDT kernel.cc:1075] Use fast generic engine


i:95


[WARNING 23-11-14 11:36:31.7256 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:31.7257 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:31.7257 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:36:33.9976 NZDT kernel.cc:1243] Loading model from path /tmp/tmpepa6yrpu/model/ with prefix ee4cd5e59e9c4e04
[INFO 23-11-14 11:36:34.0078 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:36:34.0079 NZDT kernel.cc:1075] Use fast generic engine


i:96


[WARNING 23-11-14 11:36:35.2389 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:35.2389 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:35.2389 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:36:36.8633 NZDT kernel.cc:1243] Loading model from path /tmp/tmpfoe2yqmk/model/ with prefix c59099210a884063
[INFO 23-11-14 11:36:36.8721 NZDT kernel.cc:1075] Use fast generic engine


i:97


[WARNING 23-11-14 11:36:37.9651 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:37.9652 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:37.9652 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:36:39.9106 NZDT kernel.cc:1243] Loading model from path /tmp/tmpsy5pms7g/model/ with prefix 353a99a5fc244085
[INFO 23-11-14 11:36:39.9191 NZDT kernel.cc:1075] Use fast generic engine


i:98


[WARNING 23-11-14 11:36:41.3891 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:41.3891 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:41.3891 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:36:43.9505 NZDT kernel.cc:1243] Loading model from path /tmp/tmp18xudwbr/model/ with prefix 80ed72330330426b
[INFO 23-11-14 11:36:43.9680 NZDT kernel.cc:1075] Use fast generic engine


i:99


[WARNING 23-11-14 11:36:45.3081 NZDT gradient_boosted_trees.cc:1818] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:45.3081 NZDT gradient_boosted_trees.cc:1829] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 23-11-14 11:36:45.3082 NZDT gradient_boosted_trees.cc:1843] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 23-11-14 11:36:47.6056 NZDT kernel.cc:1243] Loading model from path /tmp/tmp2r86_ljz/model/ with prefix 05c739f8e14c46db
[INFO 23-11-14 11:36:47.6175 NZDT abstract_model.cc:1311] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 23-11-14 11:36:47.6175 NZDT kernel.cc:1075] Use fast generic engine


Submission exported to Dataset/submission1.csv


# What is next

If you want to learn more about TensorFlow Decision Forests and its advanced features, you can follow the official documentation [here](https://www.tensorflow.org/decision_forests) 